packages/core/src/utils.ts

```diff
@@ -259,6 +263,143 @@ export async function* maskStreamTags(
   }
 }
 
+/**
+ * Resolve serialized zod output - This function takes the string output ot the `jsonSchemaToZod` function
+ * and instantiates the zod object correctly.
+ *
+ * @param schema - serialized zod object
+ * @returns resolved zod object
+ */
+export function resolveSerializedZodOutput(schema: string): z.ZodType {
+  // Creates and immediately executes a new function that takes 'z' as a parameter
+  // The function body is a string that returns the serialized zod schema
+  // When executed with the 'z' parameter, it reconstructs the zod schema in the current context
+  return Function('z', `"use strict";return (${schema});`)(z);
+}
+
+/**
+ * Checks if a tool is a Vercel Tool
+ * @param tool - The tool to check
+ * @returns True if the tool is a Vercel Tool, false otherwise
+ */
+export function isVercelTool(tool: any): tool is VercelTool {
+  // Checks if the tool is an object and has a 'parameters' property
+  return typeof tool === 'object' && tool !== null && 'parameters' in tool;
+}
+
+interface ToolOptions {
+  name: string;
+  runId?: string;
+  threadId?: string;
+  resourceId?: string;
+  logger: Logger;
+  description?: string;
+  mastra?: MastraPrimitives;
```

Comment:
This should be updated; we now support both
```suggestion
  mastra?: Mastra & MastraPrimitives
```